### PR TITLE
Protect: Add FreeList and PaidList pagination

### PR DESCRIPTION
**Pull request by dkmyta**
## Description
Add pagination to the current FreeList and PaidList components

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- https://github.com/Automattic/jetpack-scan-team/issues/1369

## Does this pull request change what data or activity we track or use?

- No

## Testing instructions:

* Checkout this branch using the beta tester on JN site with many threats (over 10)
* Let the initial Protect report generate the results
* Verify the the list is paginated when over 10 threats exist
* Upgrade and allow the first scan to detect the threats as well
* Verify that the list is paginated when over 10 threats exist
* Fixed or ignore all threats and proceed to the History tab
* Verify that the list is paginated when over 10 threats exist
* Ensure no regressions are introduced

## Screenshots
Desktop:
Screen Shot on 2024-09-03 at 19-08-23

Mobile:
Screen Shot on 2024-09-03 at 19-08-09

### CODE REVIEW

File: projects/plugins/protect/src/js/components/threats-list/pagination.jsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+						>
+							{ 1 }
+						</Button>
+					) }
+					<Button
+						onClick={ handlePreviousPageClick }
+						disabled={ currentPage === 1 }
+						variant={ 'secondary' }
+					>
+						{ '<' }
```
Comment (nateweller) [inline projects/plugins/protect/src/js/components/threats-list/pagination.jsx]: Suggestion: could use an icon here from `@wordpress/icons`

**Comment by nateweller** [comment]
Design wise, could we match the pagination in Jetpack Boost or CRM?

IMO this could make a great component for the Jetpack design library, if you wanted to take this task further. I'm happy to merge this directly into Protect first though.

Boost:

Screenshot 2024-08-29 at 4 19 24 PM

CRM:

Screenshot 2024-08-29 at 4 23 47 PM

File: projects/plugins/protect/src/js/components/threats-list/pagination.jsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		const indexOfLastItem = currentPage * itemPerPage;
+		const indexOfFirstItem = indexOfLastItem - itemPerPage;
+		return list.slice( indexOfFirstItem, indexOfLastItem );
+	}, [ currentPage, list, itemPerPage ] );
+
+	const onPageChange = useCallback( pageNumber => {
+		setCurrentPage( pageNumber );
+	}, [] );
+
+	const [ isSmall, setIsSmall ] = useState( window.matchMedia( '(max-width: 1220px)' ).matches );
```
Comment (nateweller) [inline projects/plugins/protect/src/js/components/threats-list/pagination.jsx]: Could `useBreakpointMatch` be useful here?

Comment (dkmyta) [reply projects/plugins/protect/src/js/components/threats-list/pagination.jsx]: I started with `useBreakpointMatch` but found that we seem to have a unique requirement with how our UI shifts in various viewports that was not exactly covered by the options it provided (`isSm`, `isMd`,` isLg`).

**Comment by dkmyta** [comment]
>Design wise, could we match the pagination in Jetpack Boost or CRM

I'll look into it!

**Comment by nateweller** [comment]
@dkmyta I found a Figma file which might be useful! TTzDk3dzct551hYWRXPFa3-fi-7790_96468

File: projects/plugins/protect/src/js/components/threats-list/styles.module.scss [hunk 3]
```diff
@@ -99,3 +99,44 @@
...
+				color: var( --jp-black );
+				background: none;
+				box-shadow: none;
+			}
+		}
+
+		&.icon-button:focus:not(:disabled)  {
+			box-shadow: none;
+		}
+	}
```
Comment (nateweller) [inline projects/plugins/protect/src/js/components/threats-list/styles.module.scss]:
```suggestion

        > svg {
            width: 1.5rem; // 24px
            height: 1.5rem; // 24px
        }
	}
```
Suggestion: smaller icon sizes to match design.

Comment (dkmyta) [reply projects/plugins/protect/src/js/components/threats-list/styles.module.scss]: Addressed via [34e9ee8](https://github.com/Automattic/jetpack/pull/39058/commits/34e9ee8a74bd0d7b13247fba544b8ca44103f8b0)

**Comment by nateweller** [comment]
Looking good and testing well :+1:

Sketched out some suggestions in https://github.com/Automattic/jetpack/pull/39249

**Review by nateweller — APPROVED** [review]
Very nice enhancement to the threats UI! Looks good works good LGTM 👍

**Pull request merged**